simulator/tools/rofi_hal.hpp

```diff
@@ -0,0 +1,51 @@
+#include <memory>
+
+namespace rofi
+{
+    namespace hal
+    {
+        class RoFI
+        {
+        public:
+            class Data;
+            class Joint;
+
+        private:
+            boost::shared_ptr< Data > rdata;
+
+        public:
+            // TODO constructors
+            explicit RoFI();
+
+            Joint getJoint( int index );
+
+            class Joint
```

Comment:
I am thinking about nesting these classes. I understand the design you have proxies (`Joint`) and actual implementation (`Data`). There will be plenty of these (joints, sensors, connectors) and I am afraid it can be messy. Maybe a flat structure could be more readable. But I am not sure.

Reply:
I was thinking about something like this:
```
namespace detail {
	struct JointData;
	struct SensorData;
}

class Joint;
class Sensor;
class RoFI;
```

Reply:
Changed as proposed and I'm satisfied with how it turned out